src/Build/Logging/TerminalLogger/TerminalLogger.cs

```diff
@@ -49,7 +49,19 @@ internal record struct ProjectContext(int Id)
     {
         public ProjectContext(BuildEventContext context)
             : this(context.ProjectContextId)
-        { }
+        {
+        }
+    }
+
+    /// <summary>
+    /// A wrapper over the evaluation context ID passed to us in <see cref="IEventSource"/> logger events.
+    /// </summary>
+    internal record struct EvalContext(int Id)
+    {
+        public EvalContext(BuildEventContext context)
+            : this(context.EvaluationId)
+        {
+        }
```

Comment:
The typed-id wave continues! It's really nice to make sure we use the correct Ids from the BuildEventContext all the time.